api/context.go

```diff
@@ -378,3 +452,38 @@ func Handle404(w http.ResponseWriter, r *http.Request) {
 	l4g.Error("%v: code=404 ip=%v", r.URL.Path, GetIpAddress(r))
 	RenderWebError(err, w, r)
 }
+
+func CreateOAuthSession(token string) (*model.Session, *model.AppError) {
```

Comment:
The OAuth2 spec suggests implementing a 'scope' which will control what the app can access with the access token but that's out of the scope (heh) of this pr. So yes, this is definitely something we want in the future